shell/circleci/setup.sh

```diff
@@ -31,6 +31,13 @@ if [[ -n $TEST_RESULTS ]]; then
   mkdir -p "$TEST_RESULTS"
 fi
 
+# run prescript if user specified to install packages etc. before tests
+if [[ -n $PRE_SETUP_SCRIPT ]]; then
+  echo "⚙️ Running setup script \"${PRE_SETUP_SCRIPT}\" (from PRE_SETUP_SCRIPT)"
+  # shellcheck source=/dev/null
+  source "${PRE_SETUP_SCRIPT}"
```

Comment:
Is there a reason we are sourcing instead of executing, e.g. doing:
```suggestion
  "${PRE_SETUP_SCRIPT}"
```
?

Sourcing means:
* this has to be a bash script
* shell variables will be inherited in the current context

Reply:
decided this is needed. We want users to be able to run it against current shell

Reply:
Removed source